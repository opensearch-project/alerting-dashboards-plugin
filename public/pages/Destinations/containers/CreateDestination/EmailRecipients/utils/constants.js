/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const RECIPIENT_TYPE = {
  EMAIL: 'email',
  EMAIL_GROUP: 'email_group',
};
