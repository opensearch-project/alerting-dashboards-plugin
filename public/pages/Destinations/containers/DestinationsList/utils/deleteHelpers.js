/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const isDeleteAllowedQuery = (type, id) => ({
  size: 0,
  query: {
    nested: {
      path: `monitor.triggers.actions`,
      query: {
        bool: {
          filter: {
            term: {
              [`monitor.triggers.actions.destination_id`]: id,
            },
          },
        },
      },
    },
  },
});
