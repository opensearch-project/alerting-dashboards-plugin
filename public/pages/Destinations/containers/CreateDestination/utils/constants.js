/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DESTINATION_TYPE } from '../../../utils/constants';

export const URL_TYPE = {
  FULL_URL: 'url',
  ATTRIBUTE_URL: 'custom_url',
};

export const CONTENT_TYPE_KEY = 'Content-Type';

const DEFAULT_CONTENT_VALUE = 'application/json';
// TODO:: Change once we have complex forms for the URL like custom webhook
export const formikInitialValues = {
  urlType: 'url',
  name: '',
  type: 'slack',
  [DESTINATION_TYPE.SLACK]: {
    url: '',
  },
  [DESTINATION_TYPE.CHIME]: {
    url: '',
  },
  [DESTINATION_TYPE.CUSTOM_HOOK]: {
    urlType: URL_TYPE.FULL_URL,
    scheme: 'HTTPS',
    method: 'POST',
    headerParams: [
      {
        key: CONTENT_TYPE_KEY,
        value: DEFAULT_CONTENT_VALUE,
      },
    ],
    queryParams: [
      {
        key: '',
        value: '',
      },
    ],
  },
  [DESTINATION_TYPE.EMAIL]: {
    emailSender: [],
    emailRecipients: [],
  },
};
