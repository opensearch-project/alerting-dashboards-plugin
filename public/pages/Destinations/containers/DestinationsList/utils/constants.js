/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DESTINATION_OPTIONS } from '../../../utils/constants';

export const DEFAULT_QUERY_PARAMS = {
  from: 0,
  search: '',
  size: 20,
  sortDirection: 'desc',
  sortField: 'name',
  type: 'ALL',
};

export const MAX_DESTINATIONS = 200;

export const staticColumns = [
  {
    field: 'name',
    name: 'Destination name',
    sortable: true,
    truncateText: true,
    textOnly: true,
    width: '100px',
  },
  {
    field: 'type',
    name: 'Destination type',
    truncateText: true,
    sortable: true,
    textOnly: true,
    width: '100px',
    render: (value) => {
      //TODO:: Convert this to proper map of text to avoid filters always
      const actionType = DESTINATION_OPTIONS.filter((item) => item.value === value);
      if (actionType.length > 0) {
        return actionType[0].text;
      } else {
        return 'Unsupported Type';
      }
    },
  },
];
