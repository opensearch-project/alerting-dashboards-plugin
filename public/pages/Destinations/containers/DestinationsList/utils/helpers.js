/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import queryString from 'query-string';
import { DEFAULT_QUERY_PARAMS } from './constants';

export const getURLQueryParams = (location) => {
  const {
    from = DEFAULT_QUERY_PARAMS.from,
    size = DEFAULT_QUERY_PARAMS.size,
    search = DEFAULT_QUERY_PARAMS.search,
    sortField = DEFAULT_QUERY_PARAMS.sortField,
    sortDirection = DEFAULT_QUERY_PARAMS.sortDirection,
    type = DEFAULT_QUERY_PARAMS.type,
  } = queryString.parse(location.search);

  return {
    from: isNaN(parseInt(from, 10)) ? DEFAULT_QUERY_PARAMS.from : parseInt(from, 10),
    size: isNaN(parseInt(size, 10)) ? DEFAULT_QUERY_PARAMS.size : parseInt(size, 10),
    search,
    sortField,
    sortDirection,
    type,
  };
};
