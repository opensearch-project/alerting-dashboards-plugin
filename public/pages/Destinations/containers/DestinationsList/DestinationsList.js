/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import {
  EuiBasicTable,
  EuiCallOut,
  EuiHorizontalRule,
  EuiSpacer,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import queryString from 'query-string';
import _ from 'lodash';
import ContentPanel from '../../../../components/ContentPanel';
import {
  EmptyDestinations,
  DestinationsActions,
  DestinationsControls,
  DeleteConfirmation,
} from '../../components/DestinationsList';
import { staticColumns } from './utils/constants';
import { getURLQueryParams } from './utils/helpers';
import { isDeleteAllowedQuery } from './utils/deleteHelpers';
import { INDEX } from '../../../../../utils/constants';
import { DESTINATION_ACTIONS, OS_NOTIFICATION_PLUGIN } from '../../../../utils/constants';
import ManageSenders from '../CreateDestination/ManageSenders';
import ManageEmailGroups from '../CreateDestination/ManageEmailGroups';
import { getAllowList } from '../../utils/helpers';
import { DESTINATION_TYPE } from '../../utils/constants';
import { backendErrorNotification } from '../../../../utils/helpers';
import NotificationsInfoCallOut from '../../components/NotificationsInfoCallOut';
import FullPageNotificationsInfoCallOut from '../../components/FullPageNotificationsInfoCallOut';

class DestinationsList extends React.Component {
  constructor(props) {
    super(props);

    const { from, size, search, sortField, sortDirection, type } = getURLQueryParams(
      props.location
    );

    this.state = {
      showDeleteConfirmation: false,
      isDestinationLoading: true,
      destinations: [],
      destinationToDelete: undefined,
      totalDestinations: 0,
      page: Math.floor(from / size),
      queryParams: {
        size,
        search,
        sortField,
        sortDirection,
        type,
      },
      selectedItems: [],
      allowList: [],
      showManageSenders: false,
      showManageEmailGroups: false,
      hasNotificationPlugin: false,
    };

    this.columns = [
      ...staticColumns,
      {
        name: 'Actions',
        width: '35px',
        actions: [
          {
            // Editing Destinations is now disabled since Destinations are deprecated
            // and will automatically be migrated to Notifications Channels
            name: 'View',
            description: 'View this destination.',
            onClick: this.handleEditDestination,
          },
        ],
      },
    ];

    this.getPlugins = this.getPlugins.bind(this);
  }

  componentDidUpdate(prevProps, prevState) {
    if (
      !_.isEqual(
        { page: prevState.page, ...prevState.queryParams },
        { page: this.state.page, ...this.state.queryParams }
      )
    ) {
      const { page, queryParams } = this.state;
      this.getDestinations(page * queryParams.size, queryParams);
    }
  }

  async componentDidMount() {
    this.getPlugins();
    const { httpClient } = this.props;
    const allowList = await getAllowList(httpClient);
    this.setState({ allowList });

    const { page, queryParams } = this.state;
    this.getDestinations(page * queryParams.size, queryParams);
  }

  async getPlugins() {
    const { httpClient } = this.props;
    try {
      const pluginsResponse = await httpClient.get('../api/alerting/_plugins');
      if (pluginsResponse.ok) {
        const plugins = pluginsResponse.resp.map((plugin) => plugin.component);
        const hasNotificationPlugin = plugins.indexOf(OS_NOTIFICATION_PLUGIN) !== -1;
        this.setState({ hasNotificationPlugin });
      } else {
        console.error('There was a problem getting plugins list');
      }
    } catch (e) {
      console.error('There was a problem getting plugins list', e);
    }
  }

  isEmailAllowed = () => {
    const { allowList } = this.state;
    return allowList.includes(DESTINATION_TYPE.EMAIL);
  };

  isDeleteAllowed = async (type, id) => {
    const { httpClient } = this.props;
    const requestBody = {
      query: isDeleteAllowedQuery(type, id),
      index: INDEX.SCHEDULED_JOBS,
    };
    const resp = await httpClient.post('../api/alerting/monitors/_search', {
      body: JSON.stringify(requestBody),
    });
    const total = _.get(resp, 'resp.hits.total.value');
    return total === 0;
  };

  handleDeleteAction = async (destinationToDelete) => {
    const { id, type } = destinationToDelete;
    const allowDelete = await this.isDeleteAllowed(type, id);
    if (allowDelete) {
      this.setState({
        showDeleteConfirmation: true,
        destinationToDelete,
      });
    } else {
      this.setState({
        destinationConsumedByOthers: true,
        destinationToDelete,
      });
      //dismiss callout after 30 Seconds
      setTimeout(
        () => this.setState({ destinationConsumedByOthers: false, destinationToDelete: null }),
        5000
      );
    }
  };

  handleDeleteDestination = async () => {
    const { id: destinationId } = this.state.destinationToDelete;
    const { httpClient, notifications } = this.props;
    try {
      const resp = await httpClient.delete(`../api/alerting/destinations/${destinationId}`);
      if (resp.ok) {
        await this.getDestinations();
      } else {
        // TODO::handle error
        //Something went wrong unable to delete if trying to delete already deleted destination
        console.log('Unable to delete destination');
        backendErrorNotification(notifications, 'delete', 'destination', resp.resp);
      }
      this.setState({
        showDeleteConfirmation: false,
        destinationToDelete: null,
      });
    } catch (e) {
      console.log('unable to delete destination', e);
    }
  };

  handleEditDestination = (destinationToEdit) => {
    this.props.history.push({
      pathname: `destinations/${destinationToEdit.id}`,
      search: `?action=${DESTINATION_ACTIONS.UPDATE_DESTINATION}`,
      state: { destinationToEdit },
    });
  };

  handleSearchChange = (e) => {
    const searchText = e.target.value;
    this.setState((state) => ({
      page: 0,
      queryParams: { ...state.queryParams, search: searchText },
    }));
  };

  handleTypeChange = (e) => {
    const type = e.target.value;
    this.setState((state) => {
      return {
        page: 0,
        queryParams: { ...state.queryParams, type },
      };
    });
  };

  getDestinations = _.debounce(
    async (from, params) => {
      this.setState({
        isDestinationLoading: true,
      });
      const { history, httpClient } = this.props;
      const queryParms = queryString.stringify({ from, ...params });
      history.replace({
        ...this.props.location,
        search: queryParms,
      });
      try {
        const resp = await httpClient.get('../api/alerting/destinations', {
          query: { from, ...params },
        });
        if (resp.ok) {
          this.setState({
            isDestinationLoading: false,
            destinations: resp.destinations,
            totalDestinations: resp.totalDestinations,
          });
        } else {
          // TODO: 'response.ok' is 'false' when there is no alerting config index in the cluster, and notification should not be shown to new Alerting users
          // backendErrorNotification(notifications, 'get', 'destinations', resp.err);
          this.setState({
            isDestinationLoading: false,
          });
        }
      } catch (err) {
        console.error('Unable to get destinations', err);
      }
    },
    500,
    { leading: true }
  );

  handlePageChange = ({ page: tablePage = {}, sort = {} }) => {
    const { index: page, size } = tablePage;
    const { field: sortField, direction: sortDirection } = sort;

    this.setState((state) => ({
      page,
      queryParams: {
        ...state.queryParams,
        size,
        sortField,
        sortDirection,
      },
    }));
  };

  handlePageClick = (page) => {
    this.setState({ page });
  };

  handleResetFilter = () => {
    this.setState((state) => ({
      ...state,
      queryParams: {
        ...state.queryParams,
        search: '',
        type: 'ALL',
      },
    }));
  };

  hideManageSendersModal = () => {
    this.setState({ showManageSenders: false });
  };

  hideManageEmailGroupsModal = () => {
    this.setState({ showManageEmailGroups: false });
  };

  render() {
    const { httpClient, notifications } = this.props;
    const {
      destinations,
      destinationToDelete,
      page,
      queryParams: { size, search, type, sortDirection, sortField },
      totalDestinations,
      isDestinationLoading,
      destinationConsumedByOthers,
      allowList,
      hasNotificationPlugin,
    } = this.state;
    const isFilterApplied = !!search || type !== 'ALL';
    const pagination = {
      pageIndex: page,
      pageSize: size,
      totalItemCount: totalDestinations,
      pageSizeOptions: [5, 10, 20, 50],
    };
    const sorting = {
      sort: {
        direction: sortDirection,
        field: sortField,
      },
    };
    return (
      <React.Fragment>
        {destinationConsumedByOthers ? (
          <EuiCallOut
            title={`Couldn't delete destination ${destinationToDelete.name}. One or more monitors uses this destination.`}
            iconType="cross"
            color="danger"
          />
        ) : null}

        {isDestinationLoading || totalDestinations > 0 || isFilterApplied ? (
          <div>
            <EuiTitle size={'l'}>
              <h3>Destinations (deprecated)</h3>
            </EuiTitle>
            <EuiSpacer size={'l'} />
            <NotificationsInfoCallOut hasNotificationPlugin={hasNotificationPlugin} />
            <ContentPanel
              bodyStyles={{ padding: 'initial' }}
              title={
                <div>
                  <EuiTitle size={'s'} style={{ paddingBottom: '0px', marginBottom: '0px' }}>
                    <h3>Destinations pending for migration</h3>
                  </EuiTitle>
                  {hasNotificationPlugin ? (
                    <EuiText
                      color={'subdued'}
                      size={'s'}
                      style={{ paddingTop: '0px', marginTop: '0px' }}
                    >
                      Destinations that are pending migration will continue to work.
                    </EuiText>
                  ) : null}
                </div>
              }
              actions={
                <DestinationsActions
                  isEmailAllowed={this.isEmailAllowed()}
                  onClickManageSenders={() => {
                    this.setState({ showManageSenders: true });
                  }}
                  onClickManageEmailGroups={() => {
                    this.setState({ showManageEmailGroups: true });
                  }}
                />
              }
            >
              <DeleteConfirmation
                isVisible={this.state.showDeleteConfirmation}
                onCancel={() => {
                  this.setState({ showDeleteConfirmation: false });
                }}
                onConfirm={this.handleDeleteDestination}
              />

              <ManageSenders
                httpClient={httpClient}
                isEmailAllowed={this.isEmailAllowed()}
                isVisible={this.state.showManageSenders}
                onClickCancel={this.hideManageSendersModal}
                onClickSave={this.hideManageSendersModal}
                notifications={notifications}
              />

              <ManageEmailGroups
                httpClient={httpClient}
                isEmailAllowed={this.isEmailAllowed()}
                isVisible={this.state.showManageEmailGroups}
                onClickCancel={this.hideManageEmailGroupsModal}
                onClickSave={this.hideManageEmailGroupsModal}
                notifications={notifications}
              />

              <DestinationsControls
                activePage={page}
                pageCount={Math.ceil(totalDestinations / size) || 1}
                search={search}
                type={type}
                onSearchChange={this.handleSearchChange}
                onTypeChange={this.handleTypeChange}
                onPageClick={this.handlePageClick}
                allowList={allowList}
              />
              <EuiHorizontalRule margin="xs" />
              <EuiBasicTable
                columns={this.columns}
                hasActions={true}
                isSelectable={true}
                items={destinations}
                pagination={pagination}
                noItemsMessage={
                  isDestinationLoading ? (
                    'Loading destinations...'
                  ) : (
                    <EmptyDestinations
                      hasNotificationPlugin={hasNotificationPlugin}
                      isFilterApplied={isFilterApplied}
                      onResetFilters={this.handleResetFilter}
                    />
                  )
                }
                onChange={this.handlePageChange}
                sorting={sorting}
              />
            </ContentPanel>
          </div>
        ) : (
          <FullPageNotificationsInfoCallOut hasNotificationPlugin={hasNotificationPlugin} />
        )}
      </React.Fragment>
    );
  }
}

DestinationsList.propTypes = {
  httpClient: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  notifications: PropTypes.object.isRequired,
};
export default DestinationsList;
