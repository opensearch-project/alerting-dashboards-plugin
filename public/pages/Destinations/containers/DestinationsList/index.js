/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import DestinationsList from './DestinationsList';

export default DestinationsList;
