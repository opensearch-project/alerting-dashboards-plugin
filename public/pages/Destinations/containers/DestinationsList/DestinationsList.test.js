/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { mount } from 'enzyme';

import DestinationsList from './DestinationsList';
import { historyMock, httpClientMock } from '../../../../../test/mocks';
import { DESTINATION_TYPE } from '../../utils/constants';

const location = {
  hash: '',
  search: '',
  state: undefined,
};

const runAllPromises = () => new Promise(setImmediate);

describe('DestinationsList', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders', async () => {
    const mockSettings = {
      defaults: {
        plugins: {
          alerting: {
            destination: {
              allow_list: Object.values(DESTINATION_TYPE),
            },
          },
        },
      },
    };

    httpClientMock.get
      .mockResolvedValueOnce({
        // Mock getAllowList
        ok: true,
        resp: mockSettings,
      })
      .mockResolvedValue({
        // Mock return in getDestinations function
        ok: true,
        destinations: [],
        totalDestinations: 0,
      });

    const wrapper = mount(
      <DestinationsList httpClient={httpClientMock} history={historyMock} location={location} />
    );

    await runAllPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  test('renders when email is disallowed', async () => {
    const mockAllowList = ['chime', 'slack', 'custom_webhook'];
    const mockSettings = {
      defaults: {
        plugins: {
          alerting: {
            destination: {
              allow_list: mockAllowList,
            },
          },
        },
      },
    };

    httpClientMock.get
      .mockResolvedValueOnce({
        // Mock getAllowList
        ok: true,
        resp: mockSettings,
      })
      .mockResolvedValue({
        // Mock return in getDestinations function
        ok: true,
        destinations: [],
        totalDestinations: 0,
      });

    const wrapper = mount(
      <DestinationsList httpClient={httpClientMock} history={historyMock} location={location} />
    );

    await runAllPromises();
    wrapper.update();

    expect(wrapper.instance().state.allowList).toEqual(mockAllowList);
    expect(wrapper).toMatchSnapshot();
  });

  test('getDestinations', async () => {
    const mockSettings = {
      defaults: {
        plugins: {
          alerting: {
            destination: {
              allow_list: Object.values(DESTINATION_TYPE),
            },
          },
        },
      },
    };

    const mockDestination = {
      id: 'id',
      type: 'test_action',
      name: 'destName',
      schema_version: 1,
      seq_no: 2,
      primary_term: 3,
      test_action: 'dummy',
    };

    httpClientMock.get
      .mockResolvedValueOnce({
        // Mock getAllowList
        ok: true,
        resp: mockSettings,
      })
      .mockResolvedValue({
        // Mock return in getDestinations function
        ok: true,
        destinations: [mockDestination],
        totalDestinations: 1,
      });

    const wrapper = mount(
      <DestinationsList httpClient={httpClientMock} history={historyMock} location={location} />
    );

    await runAllPromises();

    expect(wrapper.instance().state.totalDestinations).toBe(1);
    expect(wrapper.instance().state.destinations.length).toBe(1);
    expect(wrapper.instance().state.destinations[0].id).toBe('id');
    expect(wrapper.instance().state.destinations[0].type).toBe('test_action');
    expect(wrapper.instance().state.destinations[0].name).toBe('destName');
    expect(wrapper.instance().state.destinations[0].schema_version).toBe(1);
    expect(wrapper.instance().state.destinations[0].seq_no).toBe(2);
    expect(wrapper.instance().state.destinations[0].primary_term).toBe(3);
    expect(wrapper.instance().state.destinations[0].test_action).toBe('dummy');
  });
});
