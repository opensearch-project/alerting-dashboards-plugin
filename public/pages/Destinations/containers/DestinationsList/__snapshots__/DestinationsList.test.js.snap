// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationsList renders when Notification plugin is installed 1`] = `
<DestinationsList
  history={
    Object {
      "action": "REPLACE",
      "block": [MockFunction],
      "createHref": [MockFunction],
      "go": [MockFunction],
      "goBack": [MockFunction],
      "goForward": [MockFunction],
      "length": 0,
      "listen": [MockFunction],
      "location": Object {
        "hash": "",
        "pathname": "",
        "search": "",
        "state": undefined,
      },
      "push": [MockFunction],
      "replace": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "hash": "",
              "search": "from=0&search=&size=20&sortDirection=desc&sortField=name&type=ALL",
              "state": undefined,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
  httpClient={[MockFunction]}
  location={
    Object {
      "hash": "",
      "search": "",
      "state": undefined,
    }
  }
>
  <FullPageNotificationsInfoCallOut
    hasNotificationPlugin={false}
  >
    <EuiPanel>
      <div
        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
      >
        <EuiEmptyPrompt
          actions={
            <EuiButton
              external={true}
              fill={true}
              href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
              iconSide="right"
              iconType="popout"
              target="_blank"
            >
              View install instructions
            </EuiButton>
          }
          body={
            <EuiText>
              Destinations will be deprecated going forward. Install the Notifications plugin for a new centralized place to manage your notification channels.
            </EuiText>
          }
          title={
            <h2>
              Destinations will become channels in Notifications
            </h2>
          }
        >
          <div
            className="euiEmptyPrompt"
          >
            <EuiTitle
              size="m"
            >
              <h2
                className="euiTitle euiTitle--medium"
              >
                Destinations will become channels in Notifications
              </h2>
            </EuiTitle>
            <EuiTextColor
              color="subdued"
            >
              <span
                className="euiTextColor euiTextColor--subdued"
              >
                <EuiSpacer
                  size="m"
                >
                  <div
                    className="euiSpacer euiSpacer--m"
                  />
                </EuiSpacer>
                <EuiText>
                  <div
                    className="euiText euiText--medium"
                  >
                    <EuiText>
                      <div
                        className="euiText euiText--medium"
                      >
                        Destinations will be deprecated going forward. Install the Notifications plugin for a new centralized place to manage your notification channels.
                      </div>
                    </EuiText>
                  </div>
                </EuiText>
              </span>
            </EuiTextColor>
            <EuiSpacer
              size="l"
            >
              <div
                className="euiSpacer euiSpacer--l"
              />
            </EuiSpacer>
            <EuiButton
              external={true}
              fill={true}
              href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
              iconSide="right"
              iconType="popout"
              target="_blank"
            >
              <EuiButtonDisplay
                baseClassName="euiButton"
                element="a"
                external={true}
                fill={true}
                href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
                iconSide="right"
                iconType="popout"
                isDisabled={false}
                rel="noopener noreferrer"
                target="_blank"
              >
                <a
                  className="euiButton euiButton--primary euiButton--fill"
                  disabled={false}
                  external={true}
                  href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
                  rel="noopener noreferrer"
                  style={
                    Object {
                      "minWidth": undefined,
                    }
                  }
                  target="_blank"
                >
                  <EuiButtonContent
                    className="euiButton__content"
                    iconSide="right"
                    iconType="popout"
                    textProps={
                      Object {
                        "className": "euiButton__text",
                      }
                    }
                  >
                    <span
                      className="euiButtonContent euiButtonContent--iconRight euiButton__content"
                    >
                      <EuiIcon
                        className="euiButtonContent__icon"
                        color="inherit"
                        size="m"
                        type="popout"
                      >
                        <div>
                          EuiIconMock
                        </div>
                      </EuiIcon>
                      <span
                        className="euiButton__text"
                      >
                        View install instructions
                      </span>
                    </span>
                  </EuiButtonContent>
                </a>
              </EuiButtonDisplay>
            </EuiButton>
          </div>
        </EuiEmptyPrompt>
      </div>
    </EuiPanel>
  </FullPageNotificationsInfoCallOut>
</DestinationsList>
`;

exports[`DestinationsList renders when Notification plugin is not installed 1`] = `
<DestinationsList
  history={
    Object {
      "action": "REPLACE",
      "block": [MockFunction],
      "createHref": [MockFunction],
      "go": [MockFunction],
      "goBack": [MockFunction],
      "goForward": [MockFunction],
      "length": 0,
      "listen": [MockFunction],
      "location": Object {
        "hash": "",
        "pathname": "",
        "search": "",
        "state": undefined,
      },
      "push": [MockFunction],
      "replace": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "hash": "",
              "search": "from=0&search=&size=20&sortDirection=desc&sortField=name&type=ALL",
              "state": undefined,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
  httpClient={[MockFunction]}
  location={
    Object {
      "hash": "",
      "search": "",
      "state": undefined,
    }
  }
>
  <FullPageNotificationsInfoCallOut
    hasNotificationPlugin={false}
  >
    <EuiPanel>
      <div
        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
      >
        <EuiEmptyPrompt
          actions={
            <EuiButton
              external={true}
              fill={true}
              href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
              iconSide="right"
              iconType="popout"
              target="_blank"
            >
              View install instructions
            </EuiButton>
          }
          body={
            <EuiText>
              Destinations will be deprecated going forward. Install the Notifications plugin for a new centralized place to manage your notification channels.
            </EuiText>
          }
          title={
            <h2>
              Destinations will become channels in Notifications
            </h2>
          }
        >
          <div
            className="euiEmptyPrompt"
          >
            <EuiTitle
              size="m"
            >
              <h2
                className="euiTitle euiTitle--medium"
              >
                Destinations will become channels in Notifications
              </h2>
            </EuiTitle>
            <EuiTextColor
              color="subdued"
            >
              <span
                className="euiTextColor euiTextColor--subdued"
              >
                <EuiSpacer
                  size="m"
                >
                  <div
                    className="euiSpacer euiSpacer--m"
                  />
                </EuiSpacer>
                <EuiText>
                  <div
                    className="euiText euiText--medium"
                  >
                    <EuiText>
                      <div
                        className="euiText euiText--medium"
                      >
                        Destinations will be deprecated going forward. Install the Notifications plugin for a new centralized place to manage your notification channels.
                      </div>
                    </EuiText>
                  </div>
                </EuiText>
              </span>
            </EuiTextColor>
            <EuiSpacer
              size="l"
            >
              <div
                className="euiSpacer euiSpacer--l"
              />
            </EuiSpacer>
            <EuiButton
              external={true}
              fill={true}
              href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
              iconSide="right"
              iconType="popout"
              target="_blank"
            >
              <EuiButtonDisplay
                baseClassName="euiButton"
                element="a"
                external={true}
                fill={true}
                href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
                iconSide="right"
                iconType="popout"
                isDisabled={false}
                rel="noopener noreferrer"
                target="_blank"
              >
                <a
                  className="euiButton euiButton--primary euiButton--fill"
                  disabled={false}
                  external={true}
                  href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
                  rel="noopener noreferrer"
                  style={
                    Object {
                      "minWidth": undefined,
                    }
                  }
                  target="_blank"
                >
                  <EuiButtonContent
                    className="euiButton__content"
                    iconSide="right"
                    iconType="popout"
                    textProps={
                      Object {
                        "className": "euiButton__text",
                      }
                    }
                  >
                    <span
                      className="euiButtonContent euiButtonContent--iconRight euiButton__content"
                    >
                      <EuiIcon
                        className="euiButtonContent__icon"
                        color="inherit"
                        size="m"
                        type="popout"
                      >
                        <div>
                          EuiIconMock
                        </div>
                      </EuiIcon>
                      <span
                        className="euiButton__text"
                      >
                        View install instructions
                      </span>
                    </span>
                  </EuiButtonContent>
                </a>
              </EuiButtonDisplay>
            </EuiButton>
          </div>
        </EuiEmptyPrompt>
      </div>
    </EuiPanel>
  </FullPageNotificationsInfoCallOut>
</DestinationsList>
`;

exports[`DestinationsList renders when email is disallowed 1`] = `
<DestinationsList
  history={
    Object {
      "action": "REPLACE",
      "block": [MockFunction],
      "createHref": [MockFunction],
      "go": [MockFunction],
      "goBack": [MockFunction],
      "goForward": [MockFunction],
      "length": 0,
      "listen": [MockFunction],
      "location": Object {
        "hash": "",
        "pathname": "",
        "search": "",
        "state": undefined,
      },
      "push": [MockFunction],
      "replace": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "hash": "",
              "search": "from=0&search=&size=20&sortDirection=desc&sortField=name&type=ALL",
              "state": undefined,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
  httpClient={[MockFunction]}
  location={
    Object {
      "hash": "",
      "search": "",
      "state": undefined,
    }
  }
>
  <FullPageNotificationsInfoCallOut
    hasNotificationPlugin={false}
  >
    <EuiPanel>
      <div
        className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
      >
        <EuiEmptyPrompt
          actions={
            <EuiButton
              external={true}
              fill={true}
              href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
              iconSide="right"
              iconType="popout"
              target="_blank"
            >
              View install instructions
            </EuiButton>
          }
          body={
            <EuiText>
              Destinations will be deprecated going forward. Install the Notifications plugin for a new centralized place to manage your notification channels.
            </EuiText>
          }
          title={
            <h2>
              Destinations will become channels in Notifications
            </h2>
          }
        >
          <div
            className="euiEmptyPrompt"
          >
            <EuiTitle
              size="m"
            >
              <h2
                className="euiTitle euiTitle--medium"
              >
                Destinations will become channels in Notifications
              </h2>
            </EuiTitle>
            <EuiTextColor
              color="subdued"
            >
              <span
                className="euiTextColor euiTextColor--subdued"
              >
                <EuiSpacer
                  size="m"
                >
                  <div
                    className="euiSpacer euiSpacer--m"
                  />
                </EuiSpacer>
                <EuiText>
                  <div
                    className="euiText euiText--medium"
                  >
                    <EuiText>
                      <div
                        className="euiText euiText--medium"
                      >
                        Destinations will be deprecated going forward. Install the Notifications plugin for a new centralized place to manage your notification channels.
                      </div>
                    </EuiText>
                  </div>
                </EuiText>
              </span>
            </EuiTextColor>
            <EuiSpacer
              size="l"
            >
              <div
                className="euiSpacer euiSpacer--l"
              />
            </EuiSpacer>
            <EuiButton
              external={true}
              fill={true}
              href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
              iconSide="right"
              iconType="popout"
              target="_blank"
            >
              <EuiButtonDisplay
                baseClassName="euiButton"
                element="a"
                external={true}
                fill={true}
                href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
                iconSide="right"
                iconType="popout"
                isDisabled={false}
                rel="noopener noreferrer"
                target="_blank"
              >
                <a
                  className="euiButton euiButton--primary euiButton--fill"
                  disabled={false}
                  external={true}
                  href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#questions-about-destinations"
                  rel="noopener noreferrer"
                  style={
                    Object {
                      "minWidth": undefined,
                    }
                  }
                  target="_blank"
                >
                  <EuiButtonContent
                    className="euiButton__content"
                    iconSide="right"
                    iconType="popout"
                    textProps={
                      Object {
                        "className": "euiButton__text",
                      }
                    }
                  >
                    <span
                      className="euiButtonContent euiButtonContent--iconRight euiButton__content"
                    >
                      <EuiIcon
                        className="euiButtonContent__icon"
                        color="inherit"
                        size="m"
                        type="popout"
                      >
                        <div>
                          EuiIconMock
                        </div>
                      </EuiIcon>
                      <span
                        className="euiButton__text"
                      >
                        View install instructions
                      </span>
                    </span>
                  </EuiButtonContent>
                </a>
              </EuiButtonDisplay>
            </EuiButton>
          </div>
        </EuiEmptyPrompt>
      </div>
    </EuiPanel>
  </FullPageNotificationsInfoCallOut>
</DestinationsList>
`;
